### ISSUE TITLE
Question About Methodology used in ggeffects

### ISSUE THREAD

**Issue by swaheera** [strengejacke/ggeffects#282]
Hello!

I am interested in learning about how the ggeffects produces Partial Effects Plots.

Suppose I fit a logistic regression model : logit(y = 1) = b0 +b1x1 +b2x2 + b3x3 

 I want to plot the effect of different values of b1 on P(y = 1) .

 - Do I just "chop off" the rest of the model to get logit(y=1) = b0 +b1x1 ... and then make a plot of x1 vs p(y=1)?
 - Do I "fix" the values of x2 and x3 to the average values of x2 and x3 from the dataset and then create these plots?
 - Or is this being done some other way?

How is this being performed in the ggeffects library?

Thanks!

**Comment by swaheera** [comment]
Suppose I fit a Logistic Regression model to some data - I understand that I can now estimate the "partial effects" of a predictor on the response variable.

As an example, suppose my response variable is "employment" (unemployed = 1 vs employed = 0) and my predictor variables are "age", "income" and "gender". After fitting a Logistic Regression model to this data, I can take the partial derivative of the response variable with respect to the age variable. Now,  I can fix the "gender" as "male",  take the average "income" for all "males" in my dataset - and now, I can obtain a function that shows the effect of small changes in "age" on a male with an average income.

Have I understood this correctly? 

As I understand, this partial effect would be only applicable to males earning roughly the average salary as per my data. These partial effects might be less applicable to males earning the highest possible salary, or to females earning the average salary.
Is there some way to estimate the "true" average partial effects of age? Or is this the best you can do?

Thanks!

**Comment by strengejacke** [comment]
Yes, that's correct. Maybe I would add that if you're interested in the _strength of association_, it doesn't matter much if you fix gender to male or female. The _slope_ of age doesn't change.

See this example: https://strengejacke.github.io/ggeffects/articles/ggeffects.html#marginal-effects-at-the-mean-by-other-predictors-levels

We have `c12hour`, and three levels for education (`c172code`). As you can see, the (strength of) association between c12hour and the outcome is the same for each level of c172code - that's why the lines are in parallel.

You could condition your non-focal terms. like income, using the `condition` argument. An alternative would be to calculcate _average marginal effects_, which you can do with the [*marginaleffects*](https://vincentarelbundock.github.io/marginaleffects/) package. I suggest reading [this vignette](https://strengejacke.github.io/ggeffects/articles/introduction_marginal_effects.html), which explains the differences. I personally find average marginal effects less intuitive and not in some situations even meaningless. If you're really interested in associations based on different income levels, either user income as interaction term and add interaction as second term in `terms`, or just follow the example I showed above and add interaction as second term without interaction:

```r
model <- glm(outcome ~ age + income + gender, family = binomial)
ggpredict(model, c("age", "income")`
```

or

```r
model <- glm(outcome ~ age * income + gender, family = binomial)
ggpredict(model, c("age", "income")`
```

**Comment by swaheera** [comment]
@strengejacke strengejacke : thank you so much for your reply! In the future, are you planning on extending this package to plot the marginal effects for Survival Analysis Models (e.g. Cox-PH, AFT models)? Thank you so much!

**Comment by strengejacke** [comment]
There'a already support for (some of?) those models, see https://strengejacke.github.io/ggeffects/articles/introduction_plotmethod.html#survival-models

If you have any specific model that's not working, please open a new issue so I can track that one.

**Comment by swaheera** [comment]
Thank you so much! This is truly a wonderful R package!
